Add release notes for 0.9.6 (#2249)
scalameta/metals: website/blog/2020-11-20-lithium.md

---
author: Tomasz Godzik
title: Metals v0.9.6 - Lithium
authorURL: https://twitter.com/TomekGodzik
authorImageURL: https://github.com/tgodzik.png
---

We're happy to announce the release of Metals v0.9.6, which mostly concentrates
on adding support for the newly released Scala 2.13.4, but also adds some

> improvements for alreadye xisting features.

Reviewer: ```suggestion
improvements for already existing features.
```

Author: Tnaks! Fixed! Should have turned on the spellchecker before :sweat_smile:

  </tr>
    <tr>
    <td>Contributors</td>
    <td align="center">4</td>
  </tr>
  <tr>
    <td>Closed issues</td>
    <td align="center">2</td>
  </tr>
  <tr>
    <td>New features</td>
    <td align="center">2</td>
  </tr>
</tbody>
</table>

For full details: https://github.com/scalameta/metals/milestone/29?closed=1

Metals is a language server for Scala that works with VS Code, Vim, Emacs,
Sublime Text, Atom and Eclipse. Metals is developed at the
[Scala Center](https://scala.epfl.ch/) and [VirtusLab](https://virtuslab.com)
with the help from [Lunatech](https://lunatech.com) along with contributors from
the community.

## TL;DR

Check out [https://scalameta.org/metals/](https://scalameta.org/metals/), and
give Metals a try!

- Scala 2.13.4 support
- Show implicit conversions and classes
- Navigating stacktrace in debug console

## Show implicit conversions and classes

Last
[release](/metals/blog/2020/11/10/lithium.html#show-implicits-and-type-decorations)
introduced the option to display additional data inferred from the code to a
user. Starting with this release it's also possible to show implicit conversions
and classes, which can be enabled by the additional

> `showImplicitConversionsAndClasses` setting. Moreover, Visual Studio Code
> extensions has 3 new commands to toggle all these new options, which enables

Reviewer 2: ```suggestion
`showImplicitConversionsAndClasses` setting. Moreover, the Visual Studio Code
extension has 3 new commands to toggle all these new options, which enables
```

Author: Fixed!

users to quickly peek at the additional information about their code and then
turn it back off. All these commands can be bound to a shortcut to further
improve the user experience.

![implicits](https://i.imgur.com/k6GRgue.gif)

Any editor that allows for quickly changing these settings will also benefit
from this change, as the file's decorations are refreshed instantly upon changing
any of those settings.

## Navigating stacktrace

Previously, it was possible to navigate a stacktrace using the
`Analyze stacktrace` command which was added in the
[v0.9.4](/metals/blog/2020/09/21/lithium.html#analyze-stacktrace-command)
release. It turns out, we can reuse the same mechanism to show file links in the
`Debug Console` in Visual Studio code:

![navigate-stacktrace](https://i.imgur.com/qeitymN.gif)

This was achieved by adding additional information to the output already sent to

> the editor, so this additional file links should also be reused by an other DAP

Reviewer 2: ```suggestion
the editor, so this additional file links should also be reused by any other DAP
```

Author: Fixed
